### PR TITLE
Optional field

### LINKED ISSUE #9: Put manifest without subject fails
Once the mediaType was removed the distribution fork could not handle manifests without subjects 

See manifest below - 

````json
{
    "artifactType": "test",
    "blobs": [
        {
            "mediaType": "application/vnd.oci.image.layer.v1.tar",
            "digest": "sha256:641e4de3c3a30dd1abab107fad45eef2594e200a7f2085e648035d3efe43cf44",
            "size": 4235802,
            "urls": [
                "https://github.com/sajayantony/oras/releases/download/v0.5.4-alpha/oras_0.5.4-alpha_linux_amd64.tar.gz"
            ]
        }
    ]
}
```

Also if there are url fields should we validate that the blob exists?

### PR DESCRIPTION
**Pull request by m5i-work**
Resolves #9 

Signed-off-by: Wei Meng <wemeng@microsoft.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -27,7 +27,7 @@ require (
 	github.com/ncw/swift v1.0.47
 	github.com/opencontainers/go-digest v1.0.0
 	github.com/opencontainers/image-spec v1.0.1
-	github.com/oras-project/artifacts-spec v1.0.0-rc.1
+	github.com/oras-project/artifacts-spec v1.0.0-rc.1.0.20220707054150-eddd1d8790c9
```
[the remark this answers is not in the document]
Comment (avtakkar) [reply go.mod:30]: Seems like there is already a discussion on cutting a new tag? https://github.com/oras-project/artifacts-spec/issues/110

[the remark this answers is not in the document]
Comment (akashsinghal) [reply go.mod]: @m5i-work looks like I didn't see that you had already pushed some changes here to update to rc2 :). I have a PR here https://github.com/oras-project/distribution/pull/20. I think we can consolidate the two? (the other pr also has some changes to the logic to account for optional subject, but I didn't add the tests that you added)

[the remark this answers is not in the document]
Comment (akashsinghal) [reply go.mod]: Oh yeah good point @m5i-work! I'll make the change. I'll also add the Unit Test too

[the remark this answers is not in the document]
Comment (akashsinghal) [reply go.mod]: actually I'm just going to close my PR haha :). It's basically identical

**Review by akashsinghal — APPROVED** [review]
LGTM. I closed the other duplicate PR.

**Comment by akashsinghal** [comment]
@m5i-work I spoke with @avtakkar and we are merging all rc2 changes into the rc1 feature branch. We will then cut an rc2 release based on the rc1 branch

**Pull request merged**